front/src/reducers/main/index.ts

```diff
@@ -0,0 +1,88 @@
+import { createSlice, PayloadAction } from '@reduxjs/toolkit';
+import { Notification } from 'types';
+
+export interface MainState {
+  loading: number;
+  notifications: Array<Notification>;
+  lastInterfaceVersion: string;
+}
+
+export const mainInitialState: MainState = {
+  // Number of running task
+  loading: 0,
+  // errors
+  notifications: [],
+  lastInterfaceVersion: '',
+};
+
+const mainSlice = createSlice({
+  name: 'main',
+  initialState: mainInitialState,
+  reducers: {
+    setLoading(state) {
+      state.loading += 1;
+    },
+    setSuccess(state, action: PayloadAction<Notification | undefined>) {
+      state.loading = state.loading > 0 ? state.loading - 1 : 0;
+      if (action.payload) {
+        state.notifications.push({
+          type: action.payload.type || 'success',
+          title: action.payload.title,
+          text: action.payload.text,
+          date: action.payload.date || new Date(),
+        });
+      }
+    },
+    setWarning(state, action: PayloadAction<Notification | undefined>) {
+      state.loading = state.loading > 0 ? state.loading - 1 : 0;
+      if (action.payload) {
+        state.notifications.push({
+          type: action.payload.type || 'warning',
+          title: action.payload.title,
+          text: action.payload.text,
+          date: action.payload.date || new Date(),
+        });
+      }
+    },
+    setFailure(state, action: PayloadAction<Error | undefined>) {
+      state.loading = state.loading > 0 ? state.loading - 1 : 0;
+      if (action.payload) {
+        state.notifications.push({
+          type: 'error',
+          title: action.payload.name,
+          text: action.payload.message,
+          date: new Date(),
+        });
+      }
+    },
+    addNotification(state, action: PayloadAction<Notification>) {
```

Comment:
Can you remove this action and create an action to dispatch an info notification instead ?
```suggestion
    setInfo(state, action: PayloadAction<Notification>) {
```

Reply:
I agree. But I need some explanation because I see that addNotification can be called to handle 'info', 'success', 'error' notification. So I am asking why use another action to handle different type of notification instead of using actions setSuccess, setError, etc... I think we can change these acions name by the way. Use name more indicative of what the action does like pushSuccessNotification, pushFailureNotification,...